ansible-ipi-install/roles/scale-node-prep/tasks/main.yml

```diff
@@ -86,9 +86,32 @@
     - bootorder
 
 - name: Supermicro Boot order to PXE
-  shell: ipmitool -I lanplus -H mgmt-{{ item }} -U {{ lab_ipmi_user }} -P {{ lab_ipmi_password }} chassis bootdev pxe options=persistent
-  with_items:
-    - "{{ supermicro_nodes }}"
+  block:
+    - name: Boot order for legacy mode
+      shell: ipmitool -I lanplus -H mgmt-{{ item }} -U {{ lab_ipmi_user }} -P {{ lab_ipmi_password }} chassis bootdev pxe options=persistent
+      with_items:
+        - "{{ supermicro_nodes }}"
+      when:
+        - bootmode is defined
+        - bootmode == 'legacy'
+
+    - name: Boot order for UEFI mode
+      shell: ipmitool -I lanplus -H mgmt-{{ item }} -U {{ lab_ipmi_user }} -P {{ lab_ipmi_password }} raw 0x0 0x8 0x05 0xe0 0x04 0x0 0x0 0x0
+      with_items:
+        - "{{ supermicro_nodes }}"
+      when:
+        - ( bootmode is not defined ) or ( bootmode != 'legacy' )
```

Comment:
Do we want to default to UEFI when bootmode is not defined? I assume that is because other parts of the code that generate install-config are also doing that then?

Reply:
yes, if `bootmode` is not defined install-config also set it to be UEFI mode.  
https://github.com/openshift/installer/blob/release-4.6/docs/user/metal/install_ipi.md#install-config